qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/pairing_table_widget/pairing_table_widget_view.py

```diff
@@ -268,126 +326,80 @@ def on_cell_changed(self, row, column):
+
         # set-up the menu
-        self.menu.addAction(self.add_pair_action)
-        self.menu.addAction(self.remove_pair_action)
-        self.menu.popup(QtGui.QCursor.pos())
+        menu.addAction(add_pair_action)
+        menu.addAction(remove_pair_action)
+        menu.popup(self.cursor_position)
 
     def _context_menu_add_pair_action(self, slot):
-        add_pair_action = QtWidgets.QAction("Add Pair", self)
+        add_pair_action = self.get_add_pair_action()
         add_pair_action.setCheckable(False)
-        if len(self._get_selected_row_indices()) > 0:
+        if len(self._presenter.get_selected_row_indices()) > 0:
             add_pair_action.setEnabled(False)
         add_pair_action.triggered.connect(slot)
         return add_pair_action
 
     def _context_menu_remove_pair_action(self, slot):
-        if len(self._get_selected_row_indices()) > 1:
+        if len(self._presenter.get_selected_row_indices()) > 1:
             # use plural if >1 item selected
-            remove_pair_action = QtWidgets.QAction("Remove Pairs", self)
+            remove_pair_action = self.get_remove_pair_action(True)
         else:
-            remove_pair_action = QtWidgets.QAction("Remove Pair", self)
-        if self.num_rows() == 0:
+            remove_pair_action = self.get_remove_pair_action()
+        if self.get_pairing_table.rowCount() == 0:
             remove_pair_action.setEnabled(False)
         remove_pair_action.triggered.connect(slot)
         return remove_pair_action
 
     # ------------------------------------------------------------------------------------------------------------------
     # Adding / Removing pairs
     # ------------------------------------------------------------------------------------------------------------------
-
-    def _get_selected_row_indices(self):
-        return list(set(index.row() for index in self.pairing_table.selectedIndexes()))
-
-    def get_selected_pair_names_and_indexes(self):
-        indexes = self._get_selected_row_indices()
-        return [[str(self.pairing_table.item(i, 0).text()), i] for i in indexes]
-
-    def remove_pair_by_index(self, index):
-        self.pairing_table.removeRow(index)
-
-    def remove_last_row(self):
-        last_row = self.pairing_table.rowCount() - 1
-        if last_row >= 0:
-            self.pairing_table.removeRow(last_row)
+    def get_add_pair_action(self):
+        self.add_pair_action = QtWidgets.QAction("Add Pair", self)
+        return self.add_pair_action
+
+    def get_remove_pair_action(self, plural=False):
+        """
+        Create and return the 'Remove Pair' or 'Remove Pairs' QAction based on the plural flag.
+        """
+        if plural:
+            self.remove_pair_action = QtWidgets.QAction("Remove Pairs", self)
+            return self.remove_pair_action
+        else:
+            self.remove_pair_action = QtWidgets.QAction("Remove Pair", self)
+            return self.remove_pair_action
```

Comment:
This looks like it just could be 
```suggestion
        self.remove_pair_action = QtWidgets.QAction("Remove Pairs", self) if plural else QtWidgets.QAction("Remove Pair", self)
        return self.remove_pair_action
```